MotionPhotoReader/app/src/androidTest/java/com/google/android/libraries/motionphotoreader/MotionPhotoReaderTest.java

```diff
@@ -0,0 +1,235 @@
+package com.google.android.libraries.motionphotoreader;
+
+import android.graphics.Bitmap;
+import android.media.MediaExtractor;
+import android.os.Bundle;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.adobe.internal.xmp.XMPException;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.stubbing.Answer;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Instrumented test for MotionPhotoReader class.
+ */
+@RunWith(AndroidJUnit4.class)
+public class MotionPhotoReaderTest {
+    private static final String filename = "MVIMG_20200621_200240.jpg";
+    private static final int NUM_FRAMES = 44;
+    private static final long SEEK_AMOUNT_US = 10_000L;
+
+    /** A list of opened motion photo readers to close afterwards. */
+    private final List<Runnable> cleanup = new ArrayList<>();
+
+    @After
+    public void tearDown() {
+        for (Runnable r : cleanup) {
+            r.run();
+        }
+        cleanup.clear();
+    }
+
+    private static String selectFile(String filename) throws IOException {
+        InputStream input = InstrumentationRegistry.getInstrumentation().getContext().getResources().getAssets().open(filename);
+
+        // Convert Asset to File by copying such file to our cache directory
+        File f = new File(InstrumentationRegistry.getInstrumentation().getContext().getCacheDir() +"/" + filename);
+        writeBytesToFile(input, f);
+
+        return f.getAbsolutePath();
+    }
+
+    private static void writeBytesToFile(InputStream input, File file) throws IOException {
+        FileOutputStream fos = null;
+        try {
```

Comment:
Use try-with resource syntax to make more compact. Note that even if null is assigned to the resource, it can handle null cases. Also, use guava bytestreams to make things simpler:

```
private static void writeBytesToFile(InputStream input, File file) throws IOException {
  try (FileOutputStream fos = new FileOutputStream(file)) {
    ByteStreams.copy(input,  new FileOutputStream(file));
  }
}
```